x/incentives/keeper/distribute_test.go

```diff
@@ -92,6 +93,97 @@ func (suite *KeeperTestSuite) TestDistribute() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestDistributeToCLPools() {
+	fiveKRewardCoins := sdk.NewInt64Coin(defaultRewardDenom, 5000)
+	fifteenKRewardCoins := sdk.NewInt64Coin(defaultRewardDenom, 15000)
+
+	coinsToMint := sdk.Coins{sdk.NewInt64Coin(defaultRewardDenom, 1000000)}
+
+	tests := map[string]struct {
+		// setup
+		numPools int
+
+		// expected
+		expectErr             bool
+		expectedDistributions sdk.Coins
+	}{
+		"valid case: one poolId and gaugeId": {
+			numPools:              1,
+			expectErr:             false,
+			expectedDistributions: sdk.Coins{fiveKRewardCoins},
+		},
+
+		"valid case: multiple gaugeId and poolId": {
+			numPools:              3,
+			expectErr:             false,
+			expectedDistributions: sdk.Coins{fifteenKRewardCoins},
+		},
+	}
+
+	for _, tc := range tests {
+		// setup test
+		suite.SetupTest()
+		var gauges []types.Gauge
+
+		// prepare the minting account
+		addr := sdk.AccAddress([]byte("Gauge_Creation_Addr_"))
+		// mints coins so supply exists on chain
+		suite.FundAcc(addr, coinsToMint)
+
+		// make sure the module has enough funds
+		suite.App.BankKeeper.SendCoinsFromAccountToModule(suite.Ctx, addr, types.ModuleName, coinsToMint)
+
+		// prepare a CL Pool that creates gauge at the end of createPool
+		for i := 0; i < tc.numPools; i++ {
+			clPool := suite.PrepareConcentratedPool()
+			incParams := suite.App.IncentivesKeeper.GetParams(suite.Ctx).DistrEpochIdentifier
+			currEpoch := suite.App.EpochsKeeper.GetEpochInfo(suite.Ctx, incParams)
+
+			// get the gaugeId corresponding to the CL pool
+			gaugeId, err := suite.App.PoolIncentivesKeeper.GetPoolGaugeId(suite.Ctx, clPool.GetId(), currEpoch.Duration)
+			suite.Require().NoError(err)
+
+			// get the gauge from the gaudeId
+			gauge, err := suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, gaugeId)
+			suite.Require().NoError(err)
+
+			gauge.Coins = sdk.Coins{sdk.NewInt64Coin(defaultRewardDenom, 5000)}
```

Comment:
nit: please aim to use coin constructor as it does additional validation that oftentimes proves to be useful